### PR TITLE
Refine logic on FindRootCAForXX

### PR DESCRIPTION
**Pull request by Ruoyu-y**
* Remove duplicated code on finding root CA in istio-agent
* Make the well-known cert location a const

Signed-off-by: Ruoyu.Ying <ruoyu.ying@intel.com>

Found some duplicated code in istio-agent. So made some enhancement on that.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[x] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by Ruoyu-y** [comment]
@myidpt @howardjohn I am new to the Istio community and want to help refine the code. Can you guys help with the review? Thanks!

File: pkg/istio-agent/agent.go [hunk 1]
```diff
@@ -520,38 +493,66 @@ func fileExists(path string) bool {
...
+	// Using K8S - this is likely incorrect, may work by accident.(https://github.com/istio/istio/issues/22161)
+	if a.secOpts.PilotCertProvider == constants.CertProviderKubernetes {
+		return k8sCAPath, true, nil
+	}
+
+	// Using defaultRootCertFilePath while using the custom root certificate mounted in a well known location
+	if a.secOpts.PilotCertProvider == constants.CertProviderCustom {
+		return security.DefaultRootCertFilePath, true, nil
+	}
+
```
Comment (shankgan) [inline pkg/istio-agent/agent.go:534]: Isn' there supposed to be a case for certProvidernone?

File: pkg/istio-agent/agent.go [hunk 2]
```diff
@@ -470,45 +473,15 @@ func (a *Agent) Close() {
 // In addition it deals with the case the XDS server is on port 443, expected with a proper cert.
 // /etc/ssl/certs/ca-certificates.crt
 func (a *Agent) FindRootCAForXDS() (string, error) {
-	var rootCAPath string
-
-	if a.cfg.XDSRootCerts == security.SystemRootCerts {
-		// Special case input for root cert configuration to use system root certificates
-		return "", nil
-	} else if a.cfg.XDSRootCerts != "" {
```
Comment (shankgan) [inline pkg/istio-agent/agent.go:478]: can this option be safely removed?

File: pkg/istio-agent/agent.go [hunk 3]
```diff
@@ -520,38 +493,66 @@ func fileExists(path string) bool {
 
 // Find the root CA to use when connecting to the CA (Istiod or external).
 func (a *Agent) FindRootCAForCA() (string, error) {
-	var rootCAPath string
-
-	if a.cfg.CARootCerts == security.SystemRootCerts {
-		return "", nil
-	} else if a.cfg.CARootCerts != "" {
```
Comment (shankgan) [inline pkg/istio-agent/agent.go:527]: Is this option no longer being used?

Comment (Ruoyu-y) [reply pkg/istio-agent/agent.go:527]: I am not sure if this one is still in use. If we can confirm that this option is no longer being used, we can remove this part.

**Comment by Ruoyu-y** [comment]
@shankgan BTW,  i would also like to confirm if the case using k8s as cert provider is still valid.

**Comment by Ruoyu-y** [comment]
@shankgan Can you help with the answer? Thanks!

**Comment by shankgan** [comment]
> @shankgan BTW, i would also like to confirm if the case using k8s as cert provider is still valid.
 Could be removed some time eventually, but for now, this is still in use

**Comment by Ruoyu-y** [comment]
> > @shankgan BTW, i would also like to confirm if the case using k8s as cert provider is still valid.
> > Could be removed some time eventually, but for now, this is still in use

Thanks. Then i will keep this option and remove the one to use SystemRootCerts.

**Comment by Ruoyu-y** [comment]
@shankgan Any more comments on this?

**Pull request closed without merging**